## Problem Statement 1:  
 
In each of the following situations, state whether it is a correctly stated hypothesis testing problem and why? 
 
1. 𝐻0: 𝜇 = 25, 𝐻1: 𝜇 ≠ 25 
2. 𝐻0: 𝜎 > 10, 𝐻1: 𝜎 = 10 
3. 𝐻0: 𝑥  = 50, 𝐻1: 𝑥  ≠ 50
4. 𝐻0: 𝑝 = 0.1, 𝐻1: 𝑝 = 0.5 
5. 𝐻0: 𝑠 = 30, 𝐻1: 𝑠 > 30 
 
 

***

#### 1.𝐻0: 𝜇 = 25, 𝐻1: 𝜇 ≠ 25

The major purpose of hypothesis testing is to choose between two competing hypothesis about the value of a population parameter.
Here,
Null Hypothesis            H0: 𝜇 = 25
Alternate Hypothesis       𝐻1: 𝜇 ≠ 25
    In this hypothesis, it can be seen than that the hypothesis are correctly stated because the null hypothesis is stated 
    as  difference between the parameter(𝜇) and its estimate. Hence the hypothesis is correctly stated

#### 2. 𝐻0: 𝜎 > 10, 𝐻1: 𝜎 = 10
In this hypothesis, it can be seen than that the hypothesis are not correctly stated because the null hypothesis is stated 
    as no difference between the parameter(𝜎 ) and its estimate. Hence the hypothesis is not correctly stated

#### 3. 𝐻0: 𝑥  = 50, 𝐻1: 𝑥  ≠ 50
In this hypothesis, it can be observed that the hypothesis are not correctly stated because the null hypothesis is stated as no difference between the statistic(x) and its estimate. Hence, it is not correctly stated hypothesis

#### 4. 𝐻0: 𝑝 = 0.1, 𝐻1: 𝑝 = 0.5 
In this hypothesis, it can be observed that the hypothesis are not correctly stated because the null hypothesis is stated as no difference between the parameter(p) and its estimate. Hence, it is not correctly stated hypothesis


#### 5. 𝐻0: 𝑠 = 30, 𝐻1: 𝑠 > 30 
 In this hypothesis, it can be observed that the hypothesis are not correctly stated because the null hypothesis is stated as no difference between the statistic(s) and its estimate. Hence, it is not correctly stated hypothesis

***

## Problem Statement 2: 
 
The college bookstore tells prospective students that the average cost of its textbooks is Rs. 52 with a standard deviation of Rs. 4.50. A group of smart statistics students thinks that the average cost is higher. To test the bookstore’s claim against their alternative, the students will select a random sample of size 100. Assume that the mean from their random sample is Rs. 52.80. Perform a hypothesis test at the 5% level of significance and state your decision

***


Step-1: State the hypothesis.A group of smart statistics students thinks that the average cost 52 is higher.
    
H0: μ <=52
H1: μ >52
    
Step-2: Set up the significance level(α). It is given in the problem  as 5% (0.05).
    
    α=0.05
    
Step-3: Compute the random chance probability using z score and z-table.
 
 For this set of data: 
    𝜇=Rs.52 
    𝜎 = Rs.4.50
    X̄=Rs.52.80
    n=100
    Reject H0 if z>1.645
    
 z= (52.80-52.0) / (4.50/√100)= 1.78
 
You can look at the probability by looking at z- table and p-value associated with 1.78 is 0.9625 
i.e. p=P(z>1.78)=(1-0.9625)=0.0375.



Step-4: It is less than 0.05 so we will reject the Null hypothesis i.e. the average cost of textbooks is statistically significantly higher than the bookstores claim


***

## Problem Statement 3:  
 
A certain chemical pollutant in the Genesee River has been constant for several years with mean μ = 34 ppm (parts per million) and standard deviation σ = 8 ppm. A group of factory representatives whose companies discharge liquids into the river is now claiming that they have lowered the average with improved filtration devices. A group of environmentalists will test to see if this is true at the 1% level of significance. Assume that their sample of size 50 gives a mean of 32.5 ppm. Perform a hypothesis test at the 1% level of significance and state your decision. 
 
 

***

Step-1: State the hypothesis.
    
H0: μ =34
H1: μ ≠ 34
    
Step-2: Set up the significance level(α). It is given in the problem  as 1% (0.01).
    
    α=0.05
    
Step-3: Computing the random chance probability using z score and z-table.
 
 For this set of data: 
    𝜇=34ppm
    𝜎 = 8ppm
    X̄=32.5ppm
    n=50
    
    
 z= (32.5-34) / (8/√50)= -1.33
 
Critical value of z = ±2.58 for a two tailed test with significance level 0.01



Step-4: The computed value of z=-1.33 falls in acceptance region, and hence we accept the null hypothesis. Hence the claim that they have lowered the average discharge with improved filtration devices is true

***

## Problem Statement 4:  
 
Based on population figures and other general information on the U.S. population, suppose it has been estimated that, on average, a family of four in the U.S. spends about $1135 annually on dental expenditures. Suppose further that a regional dental association wants to test to determine if this figure is accurate for their area of country. To test this, 22 families of 4 are randomly selected from the population in that area of the country and a log is kept of the family’s dental expenditure for one year. The resulting data are given below. Assuming, that dental expenditure is normally distributed in the population, use the data and an alpha of 0.5 to test the dental association’s hypothesis. 
 
1008, 812, 1117, 1323, 1308, 1415, 831, 1021, 1287, 851, 930, 730, 699, 872, 913, 944, 954, 987, 1695, 995, 1003, 994 
 
 

***

Step1: Develop a Hypothesis and State significance level:
        H0:  μ =$1135
        H1:  μ ≠$1135
        
Step2:State significance level
    Significance level (α)is given in the problem as 0.5.
        α=0.05
    
Step3: Identify Test Statistic: 
    𝜇=$1135
    𝜎 =240.37
    X̄=$1031.32
    n=22
    
    
 z= (1031.32-1135) / (240.37/√22)= -2.02
 
Step4: Determine P-value:
        p=P(z>-2.02to z<2.02)=0.0434
 
Critical value of z = ±1.96 for a two tailed test with significance level 0.05


Step5: Compare P-value to Significance level:
        0.0434<0.05.
        Hence  Reject H0 hypothesis.Hence, the average dental expenditures for the population is not accurate for their area of the country.

***

## Problem Statement 5:  
 
In a report prepared by the Economic Research Department of a major bank the Department manager maintains that the average annual family income on Metropolis is $48,432.What do you conclude about the validity of the report if a random sample of 400 families shows and average income of 48,574 with a standard deviation of 2000? 
 

***

Step1: 
    Develop a Hypothesis and State significance level:
            H0:  μ = 48,432
            H1:  μ ≠ 48,432
        
Step2:State significance level:
    Significance level (α). It is not given in the problem and hence taking it as 5% (0.05).    
    α=0.05
    
Step3: Identify Test Statistic: 
    𝜇=48,432
    𝜎 = 2000
    X̄=$48,574
    n=400
    
    
 z= (48574-48432) / (2000/√400)= 1.42
 
Step4: Determine P-value:
        p=P(z>1.42to z<-1.42)=0.1556
 
Critical value of z = ±1.96 for a two tailed test with significance level 0.05


Step5: Compare P-value to Significance level:
        0.1556>0.05.
        Hence cannot Reject H0

***

## Problem Statement 6: 
 
Suppose that in past years the average price per square foot for warehouses in the United States has been $32.28. A national real estate investor wants to determine whether that figure has changed now. The investor hires a researcher who randomly samples 19 warehouses that are for sale across the United States and finds that the mean price per square foot is 31.67, with a standard deviation of 1.29. assume that the prices of warehouse footage are normally distributed in population. If the researcher uses a 5% level of significance, what statistical conclusion can be reached? What are the hypotheses? 

Step1: 
    Develop a Hypothesis 
            H0:  μ = 32.28
            H1:  μ ≠ 32.28
Step2:State significance level:
        Significance level (α) is not given in the problem as 5% (0.05).
    
    α=0.05
    
     
Step3: Identify Test Statistic: 
    𝜇=32.28
    𝜎 =1.29
    X̄=31.67
    n=19
    
    
 z= (31.67-32.28) / (1.29/√19)= -2.06
 
Step4: Determine P-value:
        p=P(z<-2.06to z>-2.06)=0.0394
 
 Critical value of z = ±1.96 for a two tailed test with significance level 0.05


Step-5: Compare P-value to Significance level:
        0.0394<0.05.
        Hence Reject H0.Hence the average price per square foot for warehouses in the United States has changed now.

***

## Problem Statement 7: 
 
Fill in the blank spaces in the table and draw your conclusions from it. 

         AcceptanceRegion	Sample	∝	β at μ=52	β at μ=50.5
             48.5<x̄<51.5	    10			
             48<x̄<52	        10			
             48.8<x̄<51.9	    16			
            48.42<x̄<51.58	   16			



In [1]:
import pandas as pd
data = {'Acceptance Region':['48.5<x̄<51.5', '48<x̄<52', '48.8<x̄<51.9', '48.42<x̄<51.58'],'Sample':[10,10,16,16]}
df=pd.DataFrame(data)
df.index = df.index + 1  
df

Acceptance Region  Sample
1      48.5<x̄<51.5      10
2          48<x̄<52      10
3      48.8<x̄<51.9      16
4    48.42<x̄<51.58      16

### Calculating  ∝ values for each of the above acceptance regions:

#### 1.  48.5<x̄<51.5	;μ=50, n=10,s= 2.5

∝ = P(x̄<48.50 when μ=50)+P(x̄>51.50 when μ=50) 

s/sqrt(n)=2.5/sqrt(10)=0.79

z1=(48.5-50/(2.5/sqrt(10))=(-1.90)=0.028717
z2=(51.5-50/(2.5/sqrt(10))=(1.90)=1-0.9706=0.028717
P(48.5<x̄<51.5)=0.028917+0.028917=0.057434

#### ∝= P(z< -1.90)+P(z>1.90)=0.057434

#### 2.  48<x̄<52		;μ=50, n=10,s= 2.5
∝ =P(z<48)=(48-50/(2.5/sqrt(10))=-2.53
P(z>51.50)=(52-50/(2.5/sqrt(10))=2.53
∝ = P(z< -2.53) + P(z>2.53)
∝=0.005703+0.005703=0.011406

#### ∝= P(z< -2.53)+P(z>2.53)=0.011406

#### 3.  48.8<x̄<51.9			;μ=50, n=16,s= 2.5

s= 2.5/sqrt(16)=2.5/4=0.625
∝ =P(z<48.8)=(48.8-50/(2.5/sqrt(16))=-1.92
P(z>51.9)=(51.9-50/(2.5/sqrt(16))=3.04
∝ = P(z< -1.92) + P(z>3.04)
P(z< -1.92)=0.027429
P(z>3.04)=0.001183
∝=0.027429+0.001183=0.028612

#### ∝= P(z< -1.92)+P(z>3.04)=0.028612

#### 4.  48.42<x̄<51.58				;μ=50, n=16,s= 2.5

s= 2.5/sqrt(16)=2.5/4=0.625
∝ =P(z<48.42)=(48.42-50/(2.5/sqrt(16))=-2.53
P(z>51.9)=(51.58-50/(2.5/sqrt(16))=2.53
∝ = P(z< -2.53) + P(z>2.53)
P(z< -2.53)=0.005703
P(z>2.53)=0.005703
∝=0.005703+0.005703=0.011406

#### ∝= P(z< -2.53) + P(z>2.53)=0.011406


In [2]:
df['∝']=[0.057434,0.011406,0.028612,0.011406]
df

Acceptance Region  Sample         ∝
1      48.5<x̄<51.5      10  0.057434
2          48<x̄<52      10  0.011406
3      48.8<x̄<51.9      16  0.028612
4    48.42<x̄<51.58      16  0.011406

### Calculating  β at μ=52 for each of the above acceptance regions:

#### 1.  48.5<x̄<51.5	;μ=52, n=10,s= 2.5

s/sqrt(n)=2.5/sqrt(10)=0.79

z1=(48.5-52/(2.5/sqrt(10))=(-4.43)=0.00001
z2=(51.5-52/(2.5/sqrt(10))=(-0.63)=0.264347
β = P(-4.43<=z<=-0.63)= P(z<=-0.63)-P(z<=-4.43)=0.264347-0.00001
β =0.264337


#### β= P(z<=-0.63)-P(z<=-4.43)=0.264337

#### 2.  48<x̄<52		;μ=52, n=10,s= 2.5
z1=P(z<48)=(48-52/(2.5/sqrt(10))=-5.06=0
z2=P(z>52)=(52-52/(2.5/sqrt(10))=0=0.5
β = P(-5.06<=z<=0)= P(z<=0)-P(z<=-5.06)=0.5-0
β = 0.5

#### β=  P(z<=0)-P(z<=-5.06)=0.5

#### 3.  48.8<x̄<51.9			;μ=50, n=16,s= 2.5

s= 2.5/sqrt(16)=2.5/4=0.625

z1 =P(z<48.8)=(48.8-52/(2.5/sqrt(16))=-5.12=0
z2=P(z>51.9)=(51.9-52/(2.5/sqrt(16))=0.16=0.436441

β = P(-5.12<=z<=-0.16)= P(z<=-0.16)-P(z<=-5.12)=0.436441-0
β = 0.436441

#### β=  P(z<=-0.16)-P(z<=-5.12)=0.436441


#### 4.  48.42<x̄<51.58				;μ=50, n=16,s= 2.5

s= 2.5/sqrt(16)=2.5/4=0.625

z1 =P(z<48.42)=(48.42-52/(2.5/sqrt(16))=-5.78=0
z2=P(z>51.9)=(51.58-52/(2.5/sqrt(16))=0.63=0.250792

β = P(-5.78<=z<=-0.63)= P(z<=-0.63)-P(z<=-5.78)=0.250792-0
β = 0.250792

#### ∝= P(z<=-0.63)-P(z<=-5.78)=0.250792


In [3]:
df['β at μ=52']=[0.264337,0.5,0.436441,0.250792]
df

Acceptance Region  Sample         ∝  β at μ=52
1      48.5<x̄<51.5      10  0.057434   0.264337
2          48<x̄<52      10  0.011406   0.500000
3      48.8<x̄<51.9      16  0.028612   0.436441
4    48.42<x̄<51.58      16  0.011406   0.250792

### Calculating  β at μ=50.5 for each of the above acceptance regions:

#### 1.  48.5<x̄<51.5	;μ=50.5, n=10,s= 2.5

s/sqrt(n)=2.5/sqrt(10)=0.79

z1=(48.5-50.5/(2.5/sqrt(10))=(-2.53)=0.005703
z2=(51.5-50.5/(2.5/sqrt(10))=(1.27)==1-0.102042=0.897958
β = P(-2.53<=z<=1.27)=P(z<=1.27)-P(z<=-2.53)=0.897958-0.005703=0.892255

β =0.892255



#### β=P(z<=1.27)-P(z<=-2.53)=0.892255

#### 2.  48<x̄<52		;μ=52, n=10,s= 2.5
z1=P(z<48)=(48-50.5/(2.5/sqrt(10))=-3.16=0.000789
z2=P(z>52)=(52-50.5/(2.5/sqrt(10))=1.90=0.971283
β = P(-3.16<=z<=1.90)=P(z<=1.90)-P(z<=-3.16)=0.971283-0.000789
β = 0.970494


#### β=   P(z<=1.90)-P(z<=-3.16)=0.970494

#### 3.  48.8<x̄<51.9			;μ=50, n=16,s= 2.5

s= 2.5/sqrt(16)=2.5/4=0.625

z1 =P(z<48.8)=(48.8-50.5/(2.5/sqrt(16))=-2.72=0.003264
z2=P(z>51.9)=(51.9-50.5/(2.5/sqrt(16))=2.24=0.987455

β = P(-2.72<=z<=2.24)=P(z<=2.24)-P(z<=-2.72)=0.687455-0.003264
β = 0.984191


#### β=  P(z<=2.24)-P(z<=-2.72)=0.984191


#### 4.  48.42<x̄<51.58				;μ=50, n=16,s= 2.5

s= 2.5/sqrt(16)=2.5/4=0.625

z1 =P(z<48.42)=(48.42-50.5/(2.5/sqrt(16))=-3.33=0.000437
z2=P(z>51.9)=(51.58-50.5/(2.5/sqrt(16))=1.73=0.958006

β = P(-3.33<=z<=1.73)=P(z<=1.73)-P(z<=-3.33)=0.958006-0.000437
β =0.957569


#### ∝= P(z<=1.73)-P(z<=-3.33)=0.957569


In [4]:
df['β at μ=50.5']=[0.892255,0.970494,0.984191,0.957569]
df

Acceptance Region  Sample         ∝  β at μ=52  β at μ=50.5
1      48.5<x̄<51.5      10  0.057434   0.264337     0.892255
2          48<x̄<52      10  0.011406   0.500000     0.970494
3      48.8<x̄<51.9      16  0.028612   0.436441     0.984191
4    48.42<x̄<51.58      16  0.011406   0.250792     0.957569

*** 

### Problem Statement 8:
Find the t-score for a sample size of 16 taken from a population with mean 10 when the sample mean is 12 and the sample standard deviation is 1.5.

***


Test statistic:
t- score is calculated as 
 t= (x̄-μ)/(s/√n)
     x̄ = sample mean 
    μ0 = population mean 
    s = sample standard deviation 
    n = sample size
 
 Degrees of freedom - the number of observations in a data set that are free to change without changing the mean. For a single group test 𝑑𝑓 = 𝑁 − 1 For 2-group tests 𝑑𝑓 = 𝑁1 + 𝑁2 − 2  

Assumption: The differences are treated as a random sample from a normal distribution.

We reject H0 if the t value falls in the rejection region which is based on the significance level α
and n − 1 degrees of freedom.


t= (12-10)/(1.5/√16)

t= 5.33


***

### Problem Statement 9:
Find the t-score below which we can expect 99% of sample means will fall if samples
of size 16 are taken from a normally distributed population.

***

 Degrees of freedom - the number of observations in a data set that are free to change without changing the mean. For a single group test 𝑑𝑓 = 𝑁 − 1 .The sample size is 16 and hence the degree of freedon, df=16-1=15
 
    1 − 𝛼 = 0.99
 
    𝛼 = 0.01 
    
    𝑑𝑓 = 𝑛 − 1 
    𝑡(0.99) = −𝑡(0.01) 
    𝑑𝑓 = 15
    
    𝑡(0.99) = −𝑡(0.01) = −2.602

***

### Problem Statement 10:
If a random sample of size 25 drawn from a normal population gives a mean of 60 and a standard deviation of 4, find the range of t-scores where we can expect to find the middle 95% of all sample means. Compute the probability that (−𝑡0.05 <𝑡<𝑡0.10).

***

##### Test statistic:
t- score is calculated as 
 t= (x̄-μ)/(s/√n)
    x̄ = sample mean =60
    s = sample standard deviation=4 
    n = sample size=25
    
Since t0.01 leaves an area of 0.01 to the right, and − t0.05 an area of 0.05 to the left,we find the total area to be 
1 − 0.01 − 0.05 = 0.985 between −t0.005and t0.01. Hence,P (−t0.05<T <t0.01)=0.985.


***

### Problem Statement 11:
Two-tailed test for difference between two population means.Is there evidence to conclude that the number of people travelling from Bangalore to Chennai is different from the number of people travelling from Bangalore to Hosur in a week, given the following:

Population 1: Bangalore to Chennai 

        n1 = 1200
        x1 = 452
        s1 = 212

Population 2: Bangalore to Hosur 

        n2 = 800
        x2 = 523
        s2 = 185


***

The solution to this problem takes four steps: (1) state the hypotheses, (2) formulate an analysis plan, (3) analyze sample data, and (4) interpret results.

Step1: The first step is to state the null hypothesis and an alternative hypothesis

        Null hypothesis: μ1 - μ2 =0

        Alternative hypothesis: μ1 - μ2 ≠0

Step2: Formulate an analysis plan. 
    
    For this analysis, the significance level is not given.Hence we assume the significance level as 0.05(5%).The sample size is greater than 30,so using sample data, we will conduct a two-sample z-test of the null hypothesis.
    
    
    
Step3: Analyze sample data. 

    Using sample data, we compute the standard error (SE)
    
    SE = sqrt[(s1**2/n1) + (s2***2/n2)] 
    where s1 is the standard deviation of sample 1, s2 is the standard deviation of sample 2, 
    n1 is the size of sample 1, and n2 is the size of sample 2.
    
    n1 = 1200,x1 = 452,s1 = 212; n2 = 800, x2 = 523, s2 = 185

    SE = sqrt[((212)** 2/1200)) + ((185)**2/ 800))] = sqrt(37.453 + 42.781)=sqrt(80.235)

    SE = sqrt(80.235) = 8.957


    z = [ (x1 - x2) - (μ1-μ2) ] / SE where (μ1-μ2) is the hypothesised value for the difference in the two population means
    
    z= [ (452 - 523) - 0 ] / 8.957 = -7/3.51 = -7.926
    
     Critical value of z = ±1.96 for a two tailed test with significance level 0.05

margin_of_error = z_critical * (SE)
confidence_interval = (x1-x2) - margin_of_error, (x1-x2)+ margin_of_error
confidence_interval= (-88.55613419493457, -53.44386580506542)
    
Step4: Interpret results

The z-value falls in the left-hand rejection region for any commonly used 𝛼, and the p value is very small. Therefore, we conclude that the number of people travelling from Bangalore to Chennai is different from the number of people travelling from Bangalore to Hosur in a week.



In [5]:
#Using stats doing Hypothesis testing

import numpy as np
import pandas as pd
import math
from math import sqrt
import statistics as stat
from scipy import stats
sample_mean = 452-523
pop_mean=0
confidence = 0.95
stdev = sqrt(((212)** 2/1200) + ((185)**2/ 800))

def Zscore(sample_mean, pop_mean, stdev):
    return (sample_mean - pop_mean)/stdev


z_critical = stats.norm.ppf(q = confidence + (1 - confidence) / 2)
print('z-critical value:', z_critical)                     


z = Zscore(sample_mean, pop_mean, stdev)
print("z= ",z)
p= stats.norm.cdf(z)
print("p-value associated is ",p)
margin_of_error = z_critical * (stdev)
confint = sample_mean - margin_of_error, sample_mean + margin_of_error
print('Confidence interval:', confint)
if p <0.05:
    print("We reject null hypothesis and conclude that the number of people travelling from Bangalore to Chennai is different from the number of people travelling from Bangalore to Hosur in a week")
else:
    print("we accept null hypothesis")


z-critical value: 1.959963984540054
z=  -7.926428526759299
p-value associated is  1.1276917667853246e-15
Confidence interval: (-88.55613419493457, -53.44386580506542)
We reject null hypothesis and conclude that the number of people travelling from Bangalore to Chennai is different from the number of people travelling from Bangalore to Hosur in a week


***

### Problem Statement 12:
Is there evidence to conclude that the number of people preferring Duracell battery is different from the number of people preferring Energizer battery, given the following:

Population 1: Duracell

    n1 = 100
    x1 = 308
    s1 = 84

Population 2: Energizer

    n2 = 100
    x2 = 254
    s2 = 67


Step1: The first step is to state the null hypothesis and an alternative hypothesis

        Null hypothesis: μ1 = μ2

        Alternative hypothesis: μ1 ≠ μ2

Step2: Formulate an analysis plan. 
    
    For this analysis, the significance level is not given.Hence we assume the significance level as 0.05(5%).The sample size is greater than 30,so using sample data, we will conduct a two-sample z-test of the null hypothesis.
    
    
    
Step3: Analyze sample data. 

    Using sample data, we compute the standard error (SE)
    
    SE = sqrt[(s1**2/n1) + (s2***2/n2)] 
    where s1 is the standard deviation of sample 1, s2 is the standard deviation of sample 2, 
    n1 is the size of sample 1, and n2 is the size of sample 2.
    
    n1 = 100,x1 = 308,s1 = 84; n2 =100, x2 = 254, s2 = 67

    SE = sqrt[((84)** 2/100)) + ((67)**2/ 100))] 

    SE =10.74


    z = [ (x1 - x2) -  ] / SE = [ (308 - 254) - 0 ] / 10.74 =5.025
    
Step4: Interpret results

    The z-value falls in the right-hand rejection region for any commonly used 𝛼, and the p value is very small.Therefore, we conclude that the number of people preferring Duracell battery is different from the number of people preferring Energizer battery.

In [6]:
#Using stats doing Hypothesis testing

import numpy as np
import pandas as pd
import math
from math import sqrt
import statistics as stat
from scipy import stats
sample_mean = 308-254
pop_mean=0
confidence = 0.95
stdev = sqrt(((84)** 2/100) + ((67)**2/ 100)) 

def Zscore(sample_mean, pop_mean, stdev):
    return (sample_mean - pop_mean)/stdev


z_critical = stats.norm.ppf(q = confidence + (1 - confidence) / 2)
print('z-critical value:', z_critical)                     


z = Zscore(sample_mean, pop_mean, stdev)
print("z= ",z)
p= stats.norm.cdf(z)
print("p-value associated is ",p)
margin_of_error = z_critical * (stdev)
confint = sample_mean - margin_of_error, sample_mean + margin_of_error
print('Confidence interval:', confint)
if p <0.05:
    print("We reject null hypothesis")
else:
    print("We do not reject null hypothesis and conclude that the number of people preferring Duracell battery is not different from the number of people preferring Energizer battery")


z-critical value: 1.959963984540054
z=  5.025702668336442
p-value associated is  0.9999997492037371
Confidence interval: (32.94064528887134, 75.05935471112866)
We do not reject null hypothesis and conclude that the number of people preferring Duracell battery is not different from the number of people preferring Energizer battery


***

### Problem Statement 13:
Pooled estimate of the population variance
Does the data provide sufficient evidence to conclude that average percentage increase in the price of sugar differs when it is sold at two different prices?

Population 1: Price of sugar = Rs. 27.50
    
    n1 = 14
    x1 = 0.317%
    s1 = 0.12%

Population 2: Price of sugar = Rs. 20.00

    n2 = 9
    x2 = 0.21%
    s2 = 0.11%

***

Step1: The first step is to state the null hypothesis and an alternative hypothesis

        Null hypothesis: μ1 - μ2 =0

        Alternative hypothesis: μ1 - μ2 ≠ 0

Step2: Formulate an analysis plan. 
    
    For this analysis, the significance level is not given.Hence we assume the significance level as 0.05(5%).Using t-statistics to test the sample.
    
     n1 = 14,x1 = 0.317,s1 = 0.12; n2 =9, x2 = 0.21, s2 = 0.11
    
    
Step3: Analyze sample data. 

   Standardized Test Statistic for Hypothesis Tests Concerning the Difference Between Two Population Means: Small, Independent Samples
   
  t=(x1−x2)−D0/sqrt(s**2/(1/n1+1/n2))
                
       where s**2=(n1−1)s1**2+(n2−1)s2**2/n1+n2−2

The test statistic has Student’s t-distribution with df=n1+n2−2 degrees of freedom.

The samples must be independent, the populations must be normal, and the population standard deviations must be equal. “Small” samples means that either n1<30 or n2<30.

    s=sqrt((((13)(0.12)**2)+((8)(0.11)**2)/21)((1/14)+(1/9)))=0.049685

    t = (0.317-0.21-0)/s=0.107/0.049685
    t=2.154
    
    Critical value associated with 5% significance level and 21 degree of freedom is 2.080
Step4: Interpret results

  The t-value falls in the right hand rejection region but not far from 2.080.  The p-value is therefore just less than 0.05.Hence we reject the null hypothesis and conclude that average percentage increase in the price of sugar does not differ when it is sold at two different prices 

In [7]:
#Prooving our hypothesis using stats
from scipy.stats import t
from scipy import stats

n1 = 14
x1 = 0.317
s1 = 0.12
n2 =9
x2 = 0.21
s2 = 0.11
alpha=0.025
df = (n1+n2)-2
print("Degrees of freedom= ",df)
se =0.049685
t_stat = (x1 - x2) / se
print("t_stat={:.3f} ".format(t_stat))
cv = t.ppf(1-alpha,df)
print("Critical t-value={:.3f}".format(cv))
# calculate the p-value
p = (1 - t.cdf(abs(t_stat), df)) * 2
print("p= {:.3f}".format(p))
margin_of_error = cv * (se)
confint = (x1-x2) - margin_of_error, (x1-x2)+ margin_of_error
print('Confidence interval: {:.3f} to {:.3f}'.format((x1-x2) - margin_of_error, (x1-x2)+ margin_of_error))
print("Conclusion:")
if p < 0.05:
    print('Accept null hypothesis that the means are equal  and conclude that average percentage increase in the price of sugar  slightly differ when it is sold at two different prices ')
else:
    print('Reject the null hypothesis that the means are equal.')


Degrees of freedom=  21
t_stat=2.154 
Critical t-value=2.080
p= 0.043
Confidence interval: 0.004 to 0.210
Conclusion:
Accept null hypothesis that the means are equal  and conclude that average percentage increase in the price of sugar  slightly differ when it is sold at two different prices 


***

### Problem Statement 14:
The manufacturers of compact disk players want to test whether a small price reduction is enough to increase sales of their product. Is there evidence that the small price reduction is enough to increase sales of compact disk players?

Population 1: Before reduction

    n1 = 15
    x1 = Rs. 6598
    s1 = Rs. 844
    
Population 2: After reduction 
    
    n2 = 12
    x2 = RS. 6870
    s2 = Rs. 669


***

In [8]:
#Prooving our hypothesis using stats
from scipy.stats import t
from scipy import stats

n1 = 15
x1 = 6598
s1 = 844
n2 = 12
x2 = 6870
s2 = 669
#t=(x1−x2)−D0/sqrt(s**2/(1/n1+1/n2))
#where s**2=(n1−1)s1**2+(n2−1)s2**2/n1+n2−2
alpha=0.025
df = (n1+n2)-2
print("Degrees of freedom= ",df)
se =298.97 
t_stat = (x1 - x2) / se
print("t_stat={:.3f} ".format(t_stat))
cv = t.ppf(1-alpha,df)
print("Critical t-value={:.3f}".format(cv))
# calculate the p-value
p = (1 - t.cdf(abs(t_stat), df))* 2 
print("p= {:.3f}".format(p))
margin_of_error = cv * (se)
confint = (x1-x2) - margin_of_error, (x1-x2)+ margin_of_error
print('Confidence interval: {:.3f} to {:.3f}'.format((x1-x2) - margin_of_error, (x1-x2)+ margin_of_error))
print("Conclusion:")

if t_stat<=cv:
    print('Accept null hypothesis that the means are equal and state the small price reduction is enough to increase sales of compact disk players')
else:
    print('Reject the null hypothesis that the means are equal')

Degrees of freedom=  25
t_stat=-0.910 
Critical t-value=2.060
p= 0.372
Confidence interval: -887.740 to 343.740
Conclusion:
Accept null hypothesis that the means are equal and state the small price reduction is enough to increase sales of compact disk players


***

### Problem Statement 15:
Comparisons of two population proportions when the hypothesized difference is zero Carry out a two-tailed test of the equality of banks’ share of the car loan market in
1980 and 1995.

Population 1: 1980

    n1 = 1000
    x1 = 53
    𝑝 1 = 0.53

Population 2: 1985

    n2 = 100
    x2 = 43
    𝑝 2= 0.53


***

The test statistic for the difference between two population proportions where the null hypothesis difference is zero is:
    
    z = (p̂1-p̂2-0)/sqrt(p̂(1-p̂)(1/n1+1/n2))
    
    where p̂1 = x1/n1 is the sample proportion in sample 1 and p̂2 = x2/n2 is the sample proprtion in sample 2. 
    p̂ is the combined sample proportion in both samples,considered as a single sample
    
    p̂ = (x1 + x2)/(n1 + n2)


Step1: The first step is to state the null hypothesis and an alternative hypothesis

        Null hypothesis: p1-p2 = 0
        
        Alternative hypothesis: p1-p2 ≠ 0


Step2: Formulate an analysis plan. 
    
    This is a two-tailed test of the equality of two population proportions. The significance level is not given.Hence we assume the significance level as 0.05(5%).
    
    n1 = 1000
    x1 = 53
    p̂1 = 0.53
    n2 = 100
    x2 = 43
    p̂2= 0.53

    
Step3: Analyze sample data. 

   A large-sample test for the difference between two Population proportions
   
       p̂ = x1+x2 / n1 + n2 = 0.0872   
       1-p̂=  0.9127
       
       z = (0.43 - 0.53)/sqrt((0.0872*0.9217)(1/1000 +1/100)) = -0.1/0.0296=-3.378

Step4: Interpret results
The z-value falls in the left-hand rejection region for any commonly used 𝛼, and the p value is very small.Hence we reject he null hypothesis

       

In [9]:
#Using stats doing Hypothesis testing

import numpy as np
import pandas as pd
import math
from math import sqrt
import statistics as stat
from scipy import stats
sample_mean = 0.43-0.53
pop_mean=0
confidence = 0.95
#stdev = sqrt((0.0872*0.9217)(1/1000 +1/100))
stdev=0.0296

def Zscore(sample_mean, pop_mean, stdev):
    return (sample_mean - pop_mean)/stdev


z_critical = stats.norm.ppf(q = confidence + (1 - confidence) / 2)
print('z-critical value:', z_critical)                     


z = Zscore(sample_mean, pop_mean, stdev)
print("z= ",z)
p= stats.norm.cdf(z)
print("p-value associated is ",p)
margin_of_error = z_critical * (stdev)
confint = sample_mean - margin_of_error, sample_mean + margin_of_error
print('Confidence interval:', confint)
if p <0.05:
    print("We reject null hypothesis")
else:
    print("We do not reject null hypothesis")


z-critical value: 1.959963984540054
z=  -3.3783783783783794
p-value associated is  0.0003645733881143146
Confidence interval: (-0.15801493394238564, -0.04198506605761443)
We reject null hypothesis


### Problem Statement 16:  
 
Carry out a one-tailed test to determine whether the population proportion of traveler’s check buyers who buy at least $2500 in checks when sweepstakes prizes are offered as at least 10% higher than the proportion of such buyers when no sweepstakes are on. 

Population 1: With sweepstakes 
   
       n1 = 300 
       x1 = 120 
       𝑝1= 0.40 

Population 2: No sweepstakes 
    
    n2 = 700 
    x2 = 140 
    𝑝2= 0.20 
 
 

The test statistic for the difference between two population proportions where the null hypothesis difference is zero is:
    
    z = (p̂1-p̂2-0)/sqrt((p̂1(1-p̂1)/n1)+(p̂2(1-p̂2)/n2))
    
Step1: The first step is to state the null hypothesis and an alternative hypothesis

        Null hypothesis: p1-p2 <= 0.1
        
        Alternative hypothesis: p1-p2 > 0.1


Step2: Formulate an analysis plan. 
    
    This is a one-tailed test for differences in population proportions.The significance level is not given.Hence we assume the significance level as 0.1(10%).
    
    n1 = 300 
    x1 = 120 
    p̂1 = 0.40
    n2 = 700 
    x2 = 140 
    p̂2= 0.20

    
Step3: Analyze sample data. 

   A large-sample test for the difference between two Population proportions
   
       z = (0.4 - 0.2)/sqrt(((0.4*0.6)/300)+((0.2*0.8)/700))) = 0.2/0.032071=6.236
      Considering significance level of 5%, the critical value=1.96

Step4: Interpret results
The z-value falls in the right-hand rejection region for any commonly used 𝛼, and the p value is very small.Hence we fail to reject the null hypothesis

       

In [10]:
#Using stats doing Hypothesis testing

import numpy as np
import pandas as pd
import math
from math import sqrt
import statistics as stat
from scipy import stats
mean = 0.4-0.2
pop_mean=0
confidence = 0.95
#z = (p̂1-p̂2)/sqrt((p̂1(1-p̂1)/n1)+(p̂2(1-p̂2)/n2))
z = mean/sqrt(((0.4*0.6)/300)+((0.2*0.8)/700)) 
#stdev = sqrt((0.0872*0.9217)(1/1000 +1/100))
stdev=0.032071

def Zscore(mean, pop_mean, stdev):
    return (mean - pop_mean)/stdev


z_critical = stats.norm.ppf(q = confidence + (1 - confidence) / 2)
print('z-critical value:{:.3f}'.format(z_critical))


z = Zscore(mean, pop_mean, stdev)
print("z= ",z)
p= stats.norm.cdf(z)
print("p-value associated is {:.3f}".format(p))
margin_of_error = z_critical * (stdev)
confint = mean - margin_of_error, mean + margin_of_error
print('Confidence interval: {:.3f} - {:.3f}'.format(mean - margin_of_error, mean + margin_of_error))
if p <0.05:
    print("We reject null hypothesis")
else:
    print("We do not reject null hypothesis")


z-critical value:1.960
z=  6.23616351220729
p-value associated is 1.000
Confidence interval: 0.137 - 0.263
We do not reject null hypothesis


### Problem Statement 17:  
 
A die is thrown 132 times with the following results: Number turned up: 1, 2, 3, 4, 5, 6 Frequency: 16, 20, 25, 14, 29, 28 Is the die unbiased? Consider the degrees of freedom as 𝑝  −  . 

In [11]:
data = {'Numbers turned up':[1,2,3,4,5,6], 
       'Frequency': [16,20,25,14,29,28]} 
data = pd.DataFrame(data)
data

Numbers turned up  Frequency
0                  1         16
1                  2         20
2                  3         25
3                  4         14
4                  5         29
5                  6         28

In [12]:
data.set_index('Numbers turned up',inplace=True)
data

Frequency
Numbers turned up           
1                         16
2                         20
3                         25
4                         14
5                         29
6                         28

In [13]:
data.Frequency.values.sum()

132

In [14]:
data['Observed']=data.Frequency

Considering the  die is unbiased, the probability of getting each number will be 132/6. Setting the Null hypothesis as the die is unbiased.

In [15]:
data['expected']=int(132/6)

In [16]:
data

Frequency  Observed  expected
Numbers turned up                               
1                         16        16        22
2                         20        20        22
3                         25        25        22
4                         14        14        22
5                         29        29        22
6                         28        28        22

In [17]:
data[('(O - E)**2')]=(data.Observed-data.expected)**2
data

Frequency  Observed  expected  (O - E)**2
Numbers turned up                                           
1                         16        16        22          36
2                         20        20        22           4
3                         25        25        22           9
4                         14        14        22          64
5                         29        29        22          49
6                         28        28        22          36

In [18]:
data['((O-E)**2/E)']=data.iloc[:,-1]/data.iloc[:,2]
data

Frequency  Observed  expected  (O - E)**2  ((O-E)**2/E)
Numbers turned up                                                         
1                         16        16        22          36      1.636364
2                         20        20        22           4      0.181818
3                         25        25        22           9      0.409091
4                         14        14        22          64      2.909091
5                         29        29        22          49      2.227273
6                         28        28        22          36      1.636364

In [19]:
chi_squared_stat=data['((O-E)**2/E)'].values.sum()

Considering 5% level of significance, finding the table value of χ2  corresponding to 5 degrees of freedom

In [20]:
import scipy.stats as stats
n= 6
df=n - 1
print("Degrees_of_freedom= ",df)
print("Chi-Square Statistic= {:.3f}".format(chi_squared_stat))
crit = stats.chi2.ppf(q = 0.95,df=5)   

print("Critical value= {:.3f}".format(crit))

p_value = 1 - stats.chi2.cdf(x=chi_squared_stat,  # Find the p-value
                             df=5)
print("P value= {:.3f}".format(p_value))
if chi_squared_stat< crit:
    print("\nWe reject the null Hypothesis. The calculated value of χ2 is less than the table value and hence there is no evidence against the hypothesis that die is unbiased.")
    
else:
    print("We do not reject null hypothesis")

Degrees_of_freedom=  5
Chi-Square Statistic= 9.000
Critical value= 11.070
P value= 0.109

We reject the null Hypothesis. The calculated value of χ2 is less than the table value and hence there is no evidence against the hypothesis that die is unbiased.


### Problem Statement 18:  
 
In a certain town, there are about one million eligible voters. A simple random sample of 10,000 eligible voters was chosen to study the relationship between gender and participation in the last election. The results are summarized in the following 2X2 (read two by two) contingency table: 

                    Men        Women

        Voted       2792       3591

        Not voted   1486       2131
        
 We would want to check whether being a man or a woman (columns) is independent of having voted in the last election (rows). In other words, is “gender and voting independent”? 

In [21]:
import numpy as np
import pandas as pd
df = pd.DataFrame(np.array([[2792,3591],[1486,2131]])
                          ,columns=['Men','Women'])
df.rename(index={0:'Voted',1:'Not Voted'}, inplace=True)
df

Men  Women
Voted      2792   3591
Not Voted  1486   2131

Step1: State the hypothesis
    
Null H0: Gender is independent of Voting

Alternative H1: Gender and Voting are dependent

In [22]:
df.sum(axis=0)

Men      4278
Women    5722
dtype: int64

In [23]:
df['Row_Total']=df.sum(axis=1)
df

Men  Women  Row_Total
Voted      2792   3591       6383
Not Voted  1486   2131       3617

In [24]:
df['Col_Total']=df.Men.sum(),df.Women.sum()

In [25]:
df

Men  Women  Row_Total  Col_Total
Voted      2792   3591       6383       4278
Not Voted  1486   2131       3617       5722

In [26]:
expected =  np.outer(df["Row_Total"][0:2],
                     df["Col_Total"][0:2]) / 10000

expected = pd.DataFrame(expected)

expected.columns = ['Men','Women']
expected.index = ["Voted","Not Voted"]

expected

Men      Women
Voted      2730.6474  3652.3526
Not Voted  1547.3526  2069.6474

The Chi-square table has (rows-1)*(cols-1) degrees of freedom. In the case of a 2x2 table, the number of degrees of freedom is (2-1)(2-1)=1*1=1. 

In [27]:
observed=df.iloc[0:2,0:2]

In [28]:
import scipy.stats as stats
df=1
chi_squared_stat = (((observed-expected)**2)/expected).sum().sum()
 #   We call .sum() twice: once to get the column sums an
print("Degrees_of_freedom= ",df)
print("Chi-Square Statistic= {:.3f}".format(chi_squared_stat))
crit = stats.chi2.ppf(q = 0.95,df=1)   

print("Critical value= {:.3f}".format(crit))

p_value = 1 - stats.chi2.cdf(x=chi_squared_stat,  # Find the p-value
                             df=1)
print("P value= {:.3f}".format(p_value))
if p_value<0.05:
    print("\nWe reject the null Hypothesis and hence gender is independent of voting.")
    
else:
    print("We do not reject the null hypothesis.")

Degrees_of_freedom=  1
Chi-Square Statistic= 6.660
Critical value= 3.841
P value= 0.010

We reject the null Hypothesis and hence gender is independent of voting.


In [29]:
#### An alternate method to find the Critical value,p-vale and degrees of freedom using stats is as follows:


In [30]:

stats.chi2_contingency(observed= observed)

(6.552337973968181, 0.01047464310091866, 1, array([[2730.6474, 3652.3526],
        [1547.3526, 2069.6474]]))

In [31]:
stats.chisquare(f_obs= observed,   # Array of observed counts
                f_exp= expected)   # Array of expe

Power_divergenceResult(statistic=array([3.81111287, 2.84934303]), pvalue=array([0.05091361, 0.09141128]))

In [32]:
#The above output shows the test statistic,the p-value of the test,Degrees of freedomand the expected frequencies

### Problem Statement 19:  
A sample of 100 voters are asked which of four candidates they would vote for in an election. The number supporting each candidate is given below: 

            Higgins   Reardon   White   Charlton
            
            41           19      24       16 
 
Do the data suggest that all candidates are equally popular?  
        [ Chi-Square = 14.96 , with 3 df, 𝑝< 0.05] 
 

Null hypothesis H0: Candidates are equally popular

Alternate Hypothesis, H1: Preference for Candidate is there 

In [33]:
data=pd.DataFrame(np.array([[41],[19],[24],[16]]),columns=['Observed'])

data.rename(index={0:'Higgins',1:'Reardon',2:'White',3:'Charlton'}, inplace=True)
data

Observed
Higgins         41
Reardon         19
White           24
Charlton        16

In [34]:
Total=data.Observed.sum()
print("Total = ",Total)
print("Our expected frequencies = 100/4 = "+str(100/4) +" per candidate.")

Total =  100
Our expected frequencies = 100/4 = 25.0 per candidate.


In [35]:
data['Expected']= int(100/4)
data

Observed  Expected
Higgins         41        25
Reardon         19        25
White           24        25
Charlton        16        25

In [36]:
data['(O - E)**2']=(data.Observed - data.Expected)**2
data

Observed  Expected  (O - E)**2
Higgins         41        25         256
Reardon         19        25          36
White           24        25           1
Charlton        16        25          81

In [37]:
data['(O - E)**2/ E']=data['(O - E)**2']/data.Expected

In [38]:
data

Observed  Expected  (O - E)**2  (O - E)**2/ E
Higgins         41        25         256          10.24
Reardon         19        25          36           1.44
White           24        25           1           0.04
Charlton        16        25          81           3.24

In [39]:
chi_squared_stat=data.iloc[:,-1].values.sum()
print("Chi-Squared test statistic = ",chi_squared_stat)

Chi-Squared test statistic =  14.959999999999999


Considering 5% level of significance, finding the table value of χ2  corresponding to 3(i.e.,4-1) degrees of freedom

In [40]:
import scipy.stats as stats
df=3
observed=data.iloc[:,0]
expected=data.iloc[:,1]
chi_squared_stat = (((observed-expected)**2)/expected).sum().sum()
 #   We call .sum() twice: once to get the column sums an
print("Degrees_of_freedom= ",df)
print("Chi-Square Statistic= {:.3f}".format(chi_squared_stat))
crit = stats.chi2.ppf(q = 0.95,df=3)   

print("Critical value= {:.3f}".format(crit))

p_value = 1 - stats.chi2.cdf(x=chi_squared_stat,  # Find the p-value
                             df=3)
print("P value= {:.3f}".format(p_value))
if p_value<0.05:
    print("\nWe reject the null Hypothesis and conclude that candidates are not equally popular")
    
else:
    print("We do not reject the null hypothesis.")

Degrees_of_freedom=  3
Chi-Square Statistic= 14.960
Critical value= 7.815
P value= 0.002

We reject the null Hypothesis and conclude that candidates are not equally popular


The critical value of Chi-Square for a 0.05 significance level and 3 degrees of freedom is 7.82. Our obtained Chi-Square value is bigger than this, and so we conclude that our obtained value is unlikely to have occurred merely by chance. A Chi-Square value as large as ours will occur by chance only about once in a hundred trials. It seems more reasonable to conclude that the data do indeed suggest that candidates are not equally popular. 

### Problem Statement 20: 
Children of three ages are asked to indicate their preference for three photographs of adults. Do the data suggest that there is a significant relationship between age and photograph preference? What is wrong with this study? [Chi-Square = 29.6, with 4 df: 𝑝  < 0.05]. 



                                                            Photograph
                                                            
                                                         A      B       C
                                                         
                        Age of child     5-6 yrs         18     22      20
                                         
                                         7-8 yrs         2      28      40
                                         
                                         9-10 yrs        20     10      40
                                                         
                                                         










Null Hypothesis, H0: No Significant relationship between age and photograph preference
Alternate Hypothesis, H1:Significant relationship between age and photograph preference

In [41]:
data = {'A':[18,22,20], 
       'B': [2,28,40],'C':[20,10,40]} 
data = pd.DataFrame(data)
data.rename(index= {0:'5 - 6 yrs',1:'7 - 8 yrs',2:'9 - 10 yrs'},inplace=True)
data

A   B   C
5 - 6 yrs   18   2  20
7 - 8 yrs   22  28  10
9 - 10 yrs  20  40  40

In [42]:
observed=pd.DataFrame(data)

In [43]:
data.loc['Col_Total',:]= data.sum(axis=0)
data.loc[:,'Row_Total']=data.sum(axis=1)
data

A     B     C  Row_Total
5 - 6 yrs   18.0   2.0  20.0       40.0
7 - 8 yrs   22.0  28.0  10.0       60.0
9 - 10 yrs  20.0  40.0  40.0      100.0
Col_Total   60.0  70.0  70.0      200.0

#### Calculating degrees of freedom
number of Rows = 3

number of colums=3

Degrees of Freedom = (Number of rows - 1)*(Number of columns - 1)=(3-1)*(3-1)=2*2=4

Sample size=200

In [44]:
expected =  np.outer(data.iloc[-1,:] ,
                     data.iloc[:,-1] ) /200

expected = pd.DataFrame(expected)
expected.columns = ['A','B','C','Row_Total']
expected.index = ["5-6yrs","7-8yrs","9-10yrs","Col_Total"]

expected=expected.iloc[0:3,0:3]
expected

A     B     C
5-6yrs   12.0  18.0  30.0
7-8yrs   14.0  21.0  35.0
9-10yrs  14.0  21.0  35.0

In [45]:
observed = np.array([[18,2,20], [22,28,10],[20,40,40]])
expected=np.array([[12,14,14],[18,21,21],[30,35,35]])
print("(Observed-Expected)**2 :\n ",((observed-expected)**2))
Output_array=((observed-expected)**2/expected)
print("\n(Observed-Expected)**2/Expected :\n ",((observed-expected)**2/expected))
print("\nChi_square Statistics= ",Output_array.sum())

(Observed-Expected)**2 :
  [[ 36 144  36]
 [ 16  49 121]
 [100  25  25]]

(Observed-Expected)**2/Expected :
  [[ 3.         10.28571429  2.57142857]
 [ 0.88888889  2.33333333  5.76190476]
 [ 3.33333333  0.71428571  0.71428571]]

Chi_square Statistics=  29.603174603174605


In [46]:
import scipy.stats as stats
crit = stats.chi2.ppf(q =0.95,df = 4)   
chi_squared_stat=Output_array.sum()                 
print("\nChi_square Statistics= ",Output_array.sum())
print("Critical value= {:.3f}".format(crit))
if chi_squared_stat<crit:
    print("We reject null hypothesis")
else:
    print("We fail to reject null hypothesis and hence consider the alternate hypothesis")


Chi_square Statistics=  29.603174603174605
Critical value= 9.488
We fail to reject null hypothesis and hence consider the alternate hypothesis


In [47]:
result = stats.chi2_contingency(observed, correction = False)
chisq, pvalue = result[:2]
print('Chi-Square = {}, pvalue = {}'.format(chisq, pvalue))

Chi-Square = 29.603174603174605, pvalue = 5.894582174262628e-06


The critical value of Chi-Square in the table for a 0.05 significance level and 4 d.f. is 9.488. Our obtained Chi-Square value is bigger than this: therefore we have a Chi-Square value which is so large that it would occur by chance only about once in a thousand times. It seems more reasonable to accept the alternative hypothesis, that there is a significant relationship between age of child and photograph preference.

 

### Problem Statement 21:  
 
A study of conformity using the Asch paradigm involved two conditions: one where one confederate supported the true judgement and another where no confederate gave the correct response. 

                                                Support             No support
                                       
                                 Conform          18                      40
                                     
                                 Not conform      32                      10
 Is there a significant difference between the "support" and "no support" conditions in the frequency with which individuals are likely to conform? [Chi-Square = 19.87, with 1 df: 𝑝  < 0.05].  

In [48]:
import numpy as np
import pandas as pd
df = pd.DataFrame(np.array([[18,32], [40,10]])
                          ,columns=['Support','No Support'])
df.rename(index={0:'Conform',1:'Not Conform'}, inplace=True)
df

Support  No Support
Conform           18          32
Not Conform       40          10

In [49]:
Observed=df.copy()

In [50]:
df.loc['Col_Total',:]= df.sum(axis=0)
df['Row_Total']=df.sum(axis=1)
df

Support  No Support  Row_Total
Conform         18.0        32.0       50.0
Not Conform     40.0        10.0       50.0
Col_Total       58.0        42.0      100.0

In [51]:
expected =  np.outer(df["Row_Total"][0:2],
                     df.loc["Col_Total"][0:2]) / 200

expected = pd.DataFrame(expected)

expected.columns = ['Support','No Support']
expected.index = ["Conform","Not Conform"]

expected

Support  No Support
Conform         14.5        10.5
Not Conform     14.5        10.5

In [52]:
df=(2-1)*(2-1)

In [53]:
Observed.shape

(2, 2)

In [54]:
expected.shape

(2, 2)

In [55]:
chi_squared_stat = (((Observed-expected)**2)/expected).sum().sum()
 #   We call .sum() twice: once to get the column sums and a second time to add the column sums together,
#              returning the sum of the entire 2D table.
print("Chi-Square Statistic= {:.3f}".format(chi_squared_stat))
import scipy.stats as stats
crit = stats.chi2.ppf(q = 0.95, # Find the critical value for 95% confidence*
                      df = 1)   

print("Critical value= {:.3f}".format(crit))


Chi-Square Statistic= 89.737
Critical value= 3.841


In [56]:
result = stats.chi2_contingency(Observed, correction = False)
chisq, pvalue = result[:2]
print('Chi-Square = {}, pvalue = {}'.format(chisq, pvalue))

Chi-Square = 19.868637110016422, pvalue = 8.295006717156822e-06


Our obtained value of Chi-Squared is bigger than the critical value of Chi-Squared for a 0.05 significance level. In other words, there is less than a one in a thousand chance of obtaining a Chi-Square value as big as our obtained one, merely by chance. Therefore we can conclude that there is a significant difference between the "support" and "no support" conditions, in terms of the frequency with which individuals are likely to conform.

 

### Problem Statement 22:  
 
We want to test whether short people differ with respect to their leadership qualities (Genghis Khan, Adolf Hitler and Napoleon were all stature-deprived, and how many midget MP's are there?) The following table shows the frequencies with which 43 short people and 52 tall people were categorized as "leaders", "followers" or as "unclassifiable". Is there a relationship between height and leadership qualities?  
[Chi-Square = 10.71, with 2 df: 𝑝  < 0.01]. 

                                                            Height
                                                            
                                                      Short           Tall
                                                    
                                      Leader            12              32
                                        
                                      Follower          22              14
                                        
                                      Unclassifiable    9               6




 

In [57]:
import numpy as np
import pandas as pd
df = pd.DataFrame(np.array([[12,32], [22,14],[9,6]])
                          ,columns=['Short','Tall'])
df.rename(index={0:'Leader',1:'Follower',2:'Unclassifiable'}, inplace=True)
df

Short  Tall
Leader             12    32
Follower           22    14
Unclassifiable      9     6

In [58]:
observed=df.copy()

In [59]:
df.loc['Col_Total',:]= df.sum(axis=0)
df['Row_Total']=df.sum(axis=1)
df

Short  Tall  Row_Total
Leader           12.0  32.0       44.0
Follower         22.0  14.0       36.0
Unclassifiable    9.0   6.0       15.0
Col_Total        43.0  52.0       95.0

In [60]:
expected =  np.outer(df["Row_Total"][0:3],
                     df.loc["Col_Total"][0:2]) / 95

expected = pd.DataFrame(expected)

expected.columns = ['Short','Tall']
expected.index = ['Leader','Follower','Unclassifiable']

expected

Short       Tall
Leader          19.915789  24.084211
Follower        16.294737  19.705263
Unclassifiable   6.789474   8.210526

In [61]:
nRows=3
nCols=2
DF=(nRows-1)*(nCols-1)
print("Degrees of Freedom= ",DF)
chi_squared_stat = (((observed-expected)**2)/expected).sum().sum()
print("Chi-Square Statistic= {:.3f}".format(chi_squared_stat))
import scipy.stats as stats
crit = stats.chi2.ppf(q = 0.99,df = 2)   
print("Critical value= {:.3f}".format(crit))


Degrees of Freedom=  2
Chi-Square Statistic= 10.712
Critical value= 9.210


In [62]:
#Alternate method for knowing Chi-Square and p-value using stats
result = stats.chi2_contingency(observed, correction = False)
chisq, pvalue = result[:2]
print('Chi-Square = {}, pvalue = {}'.format(chisq, pvalue))

Chi-Square = 10.712198008709638, pvalue = 0.004719280137040844


Our obtained value of Chi-Squared is bigger than the critical value of Chi-Squared for a 0.01 significance level. We would conclude that there seems to be a relationship between height and leadership qualities. Note that we can only say that there is a relationship between our two variables, not that one causes the other. There could be all kinds of explanations for such a relationship.

### Problem Statement 23:  
 
Each respondent in the Current Population Survey of March 1993 was classified as employed, unemployed, or outside the labor force. The results for men in California age 35-44 can be cross-tabulated by marital status, as follows: 

 
Men of different marital status 
                                
                                            Married      Widowed,divorced or separated    Never married
                                       
                          Employed             679           103                              114
                        
                        Unemployed              63           10                                20
                        
                        Not in labor force      42           18                                25   
                                
                                
Men of different marital status seem to have different distributions of labor force status. Or is this just chance variation? (you may assume the table results from a simple random sample.)                    


Null hypothesis,H0: marital status and employment status are not associated
    
Alternative hypothesis,H1: there is some association between marital status and employment status


In [63]:
import numpy as np
import pandas as pd
df = pd.DataFrame(np.array([[679,103,114], [63,10,20],[42,18,25]])
                          ,columns=['Married','Widowed,Divorced or separated','Never married'])
df.rename(index={0:'Employed',1:'Unemployed',2:'Not in labor force'}, inplace=True)
df

Married  Widowed,Divorced or separated  Never married
Employed                679                            103            114
Unemployed               63                             10             20
Not in labor force       42                             18             25

In [64]:
observed=df.copy()

In [65]:
df.loc['Col_Total',:]= df.sum(axis=0)
df['Row_Total']=df.sum(axis=1)
df

Married  Widowed,Divorced or separated  Never married  \
Employed              679.0                          103.0          114.0   
Unemployed             63.0                           10.0           20.0   
Not in labor force     42.0                           18.0           25.0   
Col_Total             784.0                          131.0          159.0   

                    Row_Total  
Employed                896.0  
Unemployed               93.0  
Not in labor force       85.0  
Col_Total              1074.0

In [66]:
expected =  np.outer(df["Row_Total"][0:3],
                     df.loc["Col_Total"][0:3]) / 1074

expected = pd.DataFrame(expected)

expected.columns = ['Married','Widowed,Divorced or separated','Never married']
expected.index = ['Employed','Unemployed','Not in labor force']

expected

Married  Widowed,Divorced or separated  Never married
Employed            654.063315                     109.288641     132.648045
Unemployed           67.888268                      11.343575      13.768156
Not in labor force   62.048417                      10.367784      12.583799

In [67]:
nRows=3
nCols=3
DF=(nRows-1)*(nCols-1)
print("Degrees of Freedom= ",DF)
chi_squared_stat = (((observed-expected)**2)/expected).sum().sum()
print("Chi-Square Statistic= {:.3f}".format(chi_squared_stat))
import scipy.stats as stats
crit = stats.chi2.ppf(q = 0.95,df = 4)   
print("Critical value= {:.3f}".format(crit))


Degrees of Freedom=  4
Chi-Square Statistic= 31.613
Critical value= 9.488


In [68]:
#Alternate method for knowing Chi-Square and p-value using stats
result = stats.chi2_contingency(observed, correction = False)
chisq, pvalue = result[:2]
print('Chi-Square = {}, pvalue = {}'.format(chisq, pvalue))

Chi-Square = 31.61310319407798, pvalue = 2.2949883042859918e-06


In [69]:
if pvalue<0.05:
    print("Reject the null hypothesis at the 0.05 significance level.")
else:
    print("There is a relationship between rows and columns.")



Reject the null hypothesis at the 0.05 significance level.


So, the chi-squared test statistic is about 31.61 with 4 degrees of freedom.  This gives us a p-value less than 0.05.  This is unlikely to be chance variation.  It is likely that men of different marital status have different distributions of labor force status.  (Note that we cannot give a cause-and-effect relationship here.)

***